uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/microsoftsts/AdfsLoginComponentHandler.java

```diff
@@ -38,15 +38,25 @@ public class AdfsLoginComponentHandler extends AadLoginComponentHandler {
 
     @Override
     public void handleEmailField(@NonNull final String username) {
-        final UiObject2 usernameObject = UiAutomatorUtils.obtainAllEditTextObjects(CommonUtils.FIND_UI_ELEMENT_TIMEOUT).get(0);
-        usernameObject.setText(username);
+        try {
+            UiAutomatorUtils.handleInput("userNameInput", username, CommonUtils.FIND_UI_ELEMENT_TIMEOUT_SHORT);
+        } catch (AssertionError e) {
+            final UiObject2 usernameObject = UiAutomatorUtils.obtainAllEditTextObjects(CommonUtils.FIND_UI_ELEMENT_TIMEOUT).get(0);
```

Comment:
nit: please add comment in all catch blocks - which scenario we're expecting this error.